File: server/src/main/java/com/ssafy/cafe/model/service/StaffService.java
package com.ssafy.cafe.model.service;

import java.util.List;

import com.ssafy.cafe.model.dto.Order;
import com.ssafy.cafe.model.dto.Staff;


public interface StaffService {
    /**
     * 사용자 정보를 DB에 저장한다.
     * 
     * @param user
     */
    public int join(Staff staff);

    /**
     * id, pass에 해당하는 User 정보를 반환한다.
     * 
     * @param id
     * @param pass
     * @return
     * 조회된 User 정보를 반환한다.
     */
    public Staff login(String id, String pass);
    
    
    /**
     * 해당 아이디가 이미 사용 중인지를 반환한다.
     * @param id
     * @return
     */
    public boolean isUsedId(String id);

    /**
     * id 에 해당하는 User 정보를 반환한다.
     * 
     * @param id
     * @return
     * 조회된 User 정보를 반환한다.
     */
    public Staff selectUser(String id);
    
    public List<Order> selectAllWaitingOrders();
    
    public boolean updateOrderState(int orderId);
    
    
}

File: android_staff/app/src/main/java/com/ssafy/myapplication/data/remote/StaffService.kt
package com.ssafy.myapplication.data.remote

import com.ssafy.finalproject.data.model.dto.Order
import com.ssafy.myapplication.data.model.Staff
import okhttp3.RequestBody
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path
import retrofit2.http.Query

interface StaffService {

    @POST("staff/signup")
    suspend fun join(@Body body:Staff): Boolean

    @POST("staff/login")
    suspend fun login(@Body body: Staff): Staff?

    @GET("staff/isUsed")
    suspend fun isUsedId(@Query("id") id: String): Boolean

    @GET("staff/order")
    suspend fun getAllOrders(): List<Order>

    @PUT("staff/order/done")
    suspend fun updateOrderStateDone(@Body body: RequestBody) : Boolean

    @PUT("staff/order/pickup")
    suspend fun updateOrderStatePickup(@Body body: RequestBody) : Boolean

    @GET("staff/order/{orderId}")
    suspend fun getUserIdByOrderId(@Path(value = "orderId") orderId: Int) : Int

}